File: javascript/leetcode/number-of-segments-in-a-string.js
/**
 * @param {string} s
 * @return {number}
 */
var countSegments = function(s) {
    isSeg = false
    segCount = 0
    for (var i = 0; i < s.length; i++) {
        if (s[i] == ' ') {
            if (isSeg) {segCount++}
            isSeg = false
        } else {
            isSeg = true
        }
    }
    segCount += isSeg ? 1 : 0
    return segCount
};

File: java/leetcode/number-of-segments-in-a-string.java
class Solution {
    public int countSegments(String s) {
        boolean isSeg = false;
        int segCount = 0;
        for (int i = 0; i < s.length(); i++) {
            if (s.charAt(i) == ' ') {
                if (isSeg) { segCount++; }
                isSeg = false;
            } else {
                isSeg = true;
            }
        }
        segCount += isSeg ? 1 : 0;
        return segCount;
    }
}
